### DISCUSSION TITLE
Handling nested routes

### DISCUSSION THREAD

**Discussion by alexnm** [teleporthq/teleport-code-generators#554] [Ideas]
At the moment, when doing project generation we only look at the first level routes inside the root component and we assume we have a flat hierarchy of routes in the application. 

This might be a limitation we don't want moving further so we should at least start discussing about managing multiple levels of routing in projects generators.

The problem here is that each project flavor has particular ways of implementing routing so it will be tricky to come up with a single flow for all project generators.

**Comment by vladnicula** [comment]
I'd move this as a thing to talk about after we clarify state machines and switching states in state machines. Routes are a shortcut to other states of the app, and sub routes might actually be setters for various sub states of one state, or completely different states.
